File: backend/app/experience.py
"""Shared source of truth for experience levels and their starting balances.

Mirrors web/src/lib/experience.ts. Keep the two in sync when adding or
renaming levels.
"""

from decimal import Decimal
from typing import Literal

ExperienceLevel = Literal["beginner", "intermediate", "advanced", "expert"]


class ExperienceOption:
    def __init__(
        self,
        value: ExperienceLevel,
        label: str,
        balance: str,
        starting_balance: Decimal,
        description: str,
    ):
        self.value = value
        self.label = label
        self.balance = balance
        self.starting_balance = starting_balance
        self.description = description

    def to_dict(self) -> dict[str, str]:
        return {
            "value": self.value,
            "label": self.label,
            "balance": self.balance,
            "starting_balance": str(self.starting_balance),
            "description": self.description,
        }


EXPERIENCE_OPTIONS: list[ExperienceOption] = [
    ExperienceOption(
        value="beginner",
        label="Beginner",
        balance="$100,000",
        starting_balance=Decimal("100000"),
        description="Start with more capital to learn without pressure.",
    ),
    ExperienceOption(
        value="intermediate",
        label="Intermediate",
        balance="$50,000",
        starting_balance=Decimal("50000"),
        description="A balanced starting point to build your strategy.",
    ),
    ExperienceOption(
        value="advanced",
        label="Advanced",
        balance="$25,000",
        starting_balance=Decimal("25000"),
        description="Less room for error, more room to grow.",
    ),
    ExperienceOption(
        value="expert",
        label="Expert",
        balance="$10,000",
        starting_balance=Decimal("10000"),
        description="Prove your skill with limited capital.",
    ),
]

BALANCE_MAP: dict[ExperienceLevel, Decimal] = {
    o.value: o.starting_balance for o in EXPERIENCE_OPTIONS
}

File: web/src/lib/experience.ts
/**
 * Shared source of truth for experience levels, starting balances, and the
 * copy shown in the onboarding flow / reset-account dialog. Update here,
 * not in the individual components.
 */

export type Experience = "beginner" | "intermediate" | "advanced" | "expert";

export type ExperienceOption = {
  value: Experience;
  label: string;
  balance: string;
  startingBalance: string;
  description: string;
};

export const EXPERIENCE_OPTIONS: ExperienceOption[] = [
  {
    value: "beginner",
    label: "Beginner",
    balance: "$100,000",
    startingBalance: "100000",
    description: "Start with more capital to learn without pressure.",
  },
  {
    value: "intermediate",
    label: "Intermediate",
    balance: "$50,000",
    startingBalance: "50000",
    description: "A balanced starting point to build your strategy.",
  },
  {
    value: "advanced",
    label: "Advanced",
    balance: "$25,000",
    startingBalance: "25000",
    description: "Less room for error, more room to grow.",
  },
  {
    value: "expert",
    label: "Expert",
    balance: "$10,000",
    startingBalance: "10000",
    description: "Prove your skill with limited capital.",
  },
];

export const BALANCE_MAP: Record<Experience, string> = Object.fromEntries(
  EXPERIENCE_OPTIONS.map((o) => [o.value, o.startingBalance]),
) as Record<Experience, string>;

export const getExperienceOption = (value: Experience): ExperienceOption =>
  EXPERIENCE_OPTIONS.find((o) => o.value === value) ?? EXPERIENCE_OPTIONS[0];
